### PR TITLE
test/dtpools: remove lower bound for buffer size

### PR DESCRIPTION
**Pull request by gcongiu**
## Pull Request Description
Currently the object create code in dtpools only allows `maxbufsize`  that is bigger than a certain arbitrary threshold (100000). This is to avoid that the generated buffer size for objects is always bigger than the allowed max buffer size. In this cases dtpools tries creating the object again and might never succeed, ending up in an infinite loop. Remove the lower threshold and make the code break out of the loop after a max attempt threshold is reached.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

File: test/mpi/dtpools/src/dtpools.c [hunk 1]
```diff
@@ -90,23 +90,21 @@ int DTP_pool_free(DTP_pool_s dtp)
  *
  * dtp:         datatype pool handle
  * obj:         created object handle
+ * maxbufsize:  maximum buffer size allowed
```
Comment (pavanbalaji) [inline test/mpi/dtpools/src/dtpools.c:93]: This should be a separate commit.

File: test/mpi/dtpools/src/dtpools.c [hunk 2]
```diff
@@ -90,23 +90,21 @@ int DTP_pool_free(DTP_pool_s dtp)
...
  */
 int DTP_obj_create(DTP_pool_s dtp, DTP_obj_s * obj, MPI_Aint maxbufsize)
 {
     int rc = DTP_SUCCESS;
     DTPI_pool_s *dtpi = dtp.priv;
     int max_tree_depth;
+    int max_attempts = 100; /* set the max number of object create
+                               attempts to 100 and fail if exceed */
```
Comment (pavanbalaji) [inline test/mpi/dtpools/src/dtpools.c]: Can we modify this with an environment variable?

Comment (gcongiu) [reply test/mpi/dtpools/src/dtpools.c]: sure

**Comment by gcongiu** [comment]
test:mpich/warnings
test:mpich/ch4/most

File: test/mpi/util/mtest_common.c [hunk 3]
```diff
@@ -22,12 +22,12 @@ MPI_Aint MTestDefaultMaxBufferSize()
     char *envval = NULL;
     envval = getenv("MPITEST_MAXBUFFER");
     if (envval) {
-        max_size = atol(envval);
-        if (max_size < 100000) {
-            fprintf(stderr, "MPITEST_MAXBUFFER (%s) is too small, we need at least 100MB\n",
-                    envval);
+        MPI_Aint env_max_size = atol(envval);
+        if (env_max_size > max_size) {
+            fprintf(stderr, "MPITEST_MAXBUFFER (%s) is too large\n", envval);
```
Comment (pavanbalaji) [inline test/mpi/util/mtest_common.c:27]: Isn't `MPITEST_MAXBUFFER` meant for cases where the user wants to use a larger buffer size?  This error seems to defeat that purpose.

Comment (gcongiu) [reply test/mpi/util/mtest_common.c:27]: Right, I think I may have got it backwards. My interpretation was that if the user wants to change the max buffer size this should not exceed the default max_size. I will remove this commit

**Comment by hzhou** [comment]
@gcongiu Your PR title and the description seems to describe differently.

**Comment by gcongiu** [comment]
test:mpich/ch4/most
test:mpich/warnings

**Comment by pavanbalaji** [comment]
@gcongiu I think you fixed the title and description based on Hui's comment, but in the exact opposite of what this PR is actually doing.  😃 

This PR is removing the lower bound on the maximum buffer size that the user can specify.

**Comment by gcongiu** [comment]
Looking at Hui's comment I thought I got it wrong in the first place, but actually I got it wrong the second time I looked at the code, only the comment was off.

**Comment by gcongiu** [comment]
test:mpich/warnings
test:mpich/ch4/most

**Comment by pavanbalaji** [comment]
Please don't ignore ch3 testing:

test:mpich/ch3/most

**Comment by gcongiu** [comment]
Retesting CH3

test:mpich/ch3/tcp

**Comment by gcongiu** [comment]
test:mpich/ch3/tcp
